src/vmm/src/guest_config/aarch64/mod.rs

```diff
@@ -24,56 +23,16 @@ pub struct CpuConfiguration {
 
 impl CpuConfiguration {
     /// Creates new guest CPU config based on the provided template
-    pub fn new(vcpu: &VcpuFd, template: &Option<CpuTemplateType>) -> Result<Self, Error> {
-        match template {
-            Some(ref cpu_template) => Self::with_template(vcpu, cpu_template),
-            None => Ok(Self::default()),
+    pub fn apply_template(mut self, template: &CustomCpuTemplate) -> Result<Self, Error> {
```

Comment:
nitpick: While this is logical and legal, I think using the type self when actually taking in a parameter is slightly confusing. When this function is called from `configure_system_for_boot`, it gets passed a value, it doesn't use the instance's values. Arguably not a big deal.

Reply:
Actually my original idea was to take `&mut self`, but I cherry-picked Egor's implementation from PR #3615. Let's leave it as it is for now and modify it when we feel really confusing. Thanks for good input.